sw/device/silicon_creator/manuf/lib/util.c

```diff
@@ -11,24 +11,36 @@
 #include "sw/device/lib/crypto/include/hash.h"
 
 status_t manuf_util_hash_lc_transition_token(const uint32_t *raw_token,
-                                             size_t token_size,
+                                             size_t token_size_bytes,
                                              uint64_t *hashed_token) {
   crypto_const_byte_buf_t input = {
-      .data = (uint8_t *)raw_token,
-      .len = token_size,
+      .data = (unsigned char *)raw_token,
+      .len = token_num_words * sizeof(uint32_t),
   };
   crypto_const_byte_buf_t function_name_string = {
-      .data = (uint8_t *)"",
+      .data = (unsigned char *)"",
       .len = 0,
   };
   crypto_const_byte_buf_t customization_string = {
-      .data = (uint8_t *)"LC_CTRL",
+      .data = (unsigned char *)"LC_CTRL",
       .len = 7,
   };
-  crypto_byte_buf_t output = {
-      .data = (uint8_t *)hashed_token,
-      .len = token_size,
+
+  // Create a temporary uint32_t buffer and copy the result from there to the
+  // uint64_t buffer.
+  // TODO(#16556): this is a workaround to avoid violating strict-aliasing; if
+  // we pass -fno-strict-aliasing, then we can cast `hashed_token` to `uint32_t
+  // *` instead.
```

Comment:
For now I've reverted the manufacturing lib changes and added this slightly silly memcpy with a TODO, so that once #16556 is resolved we'll know to revisit it.